vc-generator/generators.js

```diff
@@ -0,0 +1,120 @@
+/*!
+ * Copyright 2023 Digital Bazaar, Inc. All Rights Reserved
+ */
+import * as vc from '@digitalbazaar/vc';
+import {DataIntegrityProof} from '@digitalbazaar/data-integrity';
+import {documentLoader} from './documentLoader.js';
+import {
+  cryptosuite as eddsa2022CryptoSuite
+} from '@digitalbazaar/eddsa-2022-cryptosuite';
+import {invalidCreateProof} from './helpers.js';
+import jsigs from 'jsonld-signatures';
+import {klona} from 'klona';
+
+const {AuthenticationProofPurpose} = jsigs.purposes;
+const {CredentialIssuancePurpose} = vc;
+
+export const vcGenerators = new Map([
+  ['issuedVc', _issuedVc],
+  ['invalidDomain', _invalidDomain],
+  ['invalidChallenge', _invalidChallenge],
+  ['invalidProofType', _incorrectProofType],
+  ['noCreated', _noCreated],
+  ['invalidCreated', _invalidCreated],
+  ['vcCreatedOneYearAgo', _vcCreatedOneYearAgo],
+  ['noVm', _noVm],
+  ['invalidVm', _invalidVm],
+  ['noProofPurpose', _noProofPurpose],
+  ['invalidProofPurpose', _invalidProofPurpose]
+]);
+
+async function _invalidProofPurpose({signer, credential}) {
+  const suite = _createEddsa2022Suite({signer});
+  suite.createProof = invalidCreateProof({mockPurpose: 'invalidPurpose'});
+  return _issueCloned({suite, credential});
+}
+
+async function _invalidDomain({signer, credential}) {
+  const suite = _createEddsa2022Suite({signer});
+  const domain = 'invalid-vc-domain.example.com';
+  const challenge = '1235abcd6789';
+  const purpose = new AuthenticationProofPurpose({challenge, domain});
+  return _issueCloned({suite, credential, purpose});
+}
+
+async function _invalidChallenge({signer, credential}) {
+  const suite = _createEddsa2022Suite({signer});
+  const domain = 'domain.example';
```

Comment:
do use `example.com` just to be sure. Maybe `invalid-challenge.example.com`

Reply:
@aljones15 this is fine. see - https://github.com/w3c-ccg/data-integrity-test-suite-assertion/pull/13#discussion_r1262841015